mlos_bench/mlos_bench/config/schemas/schedulers/scheduler-schema.json

```diff
@@ -0,0 +1,100 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://raw.githubusercontent.com/microsoft/MLOS/main/mlos_bench/mlos_bench/config/schemas/schedulers/scheduler-schema.json",
+    "title": "mlos_bench Scheduler config",
+
+    "$defs": {
+        "comment": {
+            "$comment": "This section contains reusable partial schema bits (or just split out for readability)"
+        },
+
+        "config_base_scheduler": {
+            "$comment": "config properties common to all Scheduler types.",
+            "type": "object",
+            "properties": {
+                "experiment_id": {
```

Comment:
Why is experiment_id being repeated in scheduler?
Isn't that an expected global or cli variable that's fed into other things?

Same for config_id, trial_id, etc.

Guess I think of those as controlling a run of a given instance of `mlos_bench` whereas configs for the scheduler have to do with how it schedules *new* config (e.g., how many repeats, which order, which workers, etc.).  And that in general we would have reusable scheduler configs that don't include any of those details.

Or is this just to support the possibility of one-off overrides to influence a single instance run to only consider a single pre-existing config or only invoke a single pre-existing trial (both of which are only relevant under an existing experiment_id which we would also need to reference)?

Reply:
All these parameters (except of the `trial_config_repeat_count`, probably) are mostly placeholders for properties that we need, e.g., for storage initialization - but also, yes, to make sure they are present so that other components have them, too. I like the idea of one-off overrides - I guess it should work this way. I'll add another PR with that scenario + unit tests

Reply:
> but also, yes, to make sure they are present so that other components have them, too

I don't think the need to be here for that to happen, but if we're going to implement the override option as well, it might make sense to keep them.

One general trend I see is that sometimes, because a parameter flows from, globals, overring the cli, and also appears in scheduler, and the launcher, and ...
it just gets hard to track what's *supposed* to be authoritative.

If we instead include it only in certain places and then flow thru only in the code where necessary it might be more manageable, from a config writing/debugging standpoint for users.